sql/select.go

```diff
@@ -432,6 +429,10 @@ func (v *indexInfo) makeConstraints(exprs []parser.Exprs) {
 					if !endDone {
 						constraint.end = c
 					}
+				case parser.NE:
```

Comment:
isn't this only correct for inequality to NULL? in other cases we still need to scan on either side of the value, but perhaps that is handled elsewhere